internal/gadget-context/gadget-context.go

```diff
@@ -45,80 +44,55 @@ type GadgetRunner struct {
 	resultError  error
 }
 
-func NewGadgetRunner(
+func New(
 	ctx context.Context,
 	id string,
 	runtime runtime.Runtime,
 	gadget gadgets.Gadget,
+	gadgetParams *params.Params,
 	parser parser.Parser,
 	logger logger.Logger,
-) *GadgetRunner {
-	return &GadgetRunner{
-		ctx:       ctx,
-		id:        id,
-		gadget:    gadget,
-		runtime:   runtime,
-		parser:    parser,
-		logger:    logger,
-		operators: operators.GetOperatorsForGadget(gadget),
+) *GadgetContext {
+	return &GadgetContext{
+		ctx:          ctx,
+		id:           id,
+		runtime:      runtime,
+		gadget:       gadget,
+		gadgetParams: gadgetParams,
+		parser:       parser,
+		logger:       logger,
+		operators:    operators.GetOperatorsForGadget(gadget),
```

Comment:
What to you think about passing them in the constructor? (In that way `GadgetContext` will become just a struct that holds values).

Reply:
```golang
func (r *GadgetContext) OperatorsPerGadgetParamCollection() params.Collection {
	return r.operatorsPerGadgetParamCollection
}
```

Instead of this then, we could actually name it OperatorParamCollection(), I think. And maybe instead of `PerGadgetParams()` in operators we go to `ParamDescs()` - we also have the `GlobalParamDescs()` so it should be clear that these are not global (and that can be reinforced by proper documentation). Wdyt?

Reply:
This was meant to be an answer to the comment above, sorry ;)

Reply:
I think having `ParamDescs()` and `GlobalParamDescs()` is better than having `PerGadget`.